code/modules/GOAP/monkey/info_provider.dm

```diff
@@ -0,0 +1,72 @@
+/datum/goap_info_provider/monkey/GetWorldState(datum/goap_agent/agent)
+	. = list()
+	.["GUN"] = FALSE
+	.["attackEnemy"] = FALSE
+	.["disposeEnemy"] = FALSE
+	.["Flee"] = FALSE
+	.["disarmAvailable"] = FALSE
+	.["enemyGrabbed"] = FALSE
+	.["replaceItem"] = FALSE
+	var/mob/living/carbon/monkey/C = agent.agent
+	var/obj/item/W = locate() in C.held_items
+	if(W)
+		.["hasItem"] = TRUE
+		.["hasHumanItem"] = TRUE
+		.["grabbedWeapon"] = TRUE
+		if(isgun(W))
+			var/obj/item/gun/G = W
+			if(G && G.chambered && G.chambered.BB) // we got a loaded gun!
+				.["GUN"] = TRUE
+	else
+		C.best_force = 0
+		.["hasItem"] = FALSE
+		.["hasHumanItem"] = FALSE
+		.["grabbedWeapon"] = FALSE
+	if(C.pulling)
+		.["enemyGrabbed"] = TRUE
+	var/mob/living/carbon/human/H = locate(/mob/living/carbon/human/) in view(4, agent.agent)
+	if(H && H.held_items)
+		if(prob(MONKEY_ATTACK_DISARM_PROB)) // this runs more than monkeys handle_combat did
+			.["disarmAvailable"] = TRUE
+
+/datum/goap_info_provider/monkey/GetGoal(datum/goap_agent/agent)
+	. = list()
+	var/mob/living/carbon/monkey/C = agent.agent
+	var/obj/item/W = locate() in C.held_items
+	var/itemobtainable = FALSE
+	if(length(C.enemies))
+		var/list/around = view(MONKEY_ENEMY_VISION, C)
+		for(var/mob/living/L in around)
+			if(C.should_target(L))
+				if(!L.stat) // conscious
+					if(C.health < MONKEY_FLEE_HEALTH)
+						.["Flee"] = TRUE
+						return // we don't care about any goals other than this
+					.["attackEnemy"] = TRUE // we have a target
+					break
+				else if(locate(/obj/machinery/disposal) in around)
+					if(!L.pulledby) // nobody else has got rid of them and we're not in combat
+						.["disposeEnemy"] = TRUE // they're not conscious, lets trash them will automatically grab
+						return // don't worry about getting items if we're just disposalling someone
+	var/list/itemview = oview(4, agent.agent)
```

Comment:
Only do oview() if we don't have a gun/no weapon at all, optimization for cases where oview() would be unnecessary if a gun is already equipped